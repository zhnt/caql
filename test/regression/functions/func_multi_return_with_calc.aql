// Multi-return value tests with calculations

// Test 1: Two return values with calculations
function calculate_pair(x, y) {
    let sum = x + y
    let product = x * y
    return sum, product
}

// Test 2: Three return values with string operations
function process_string(text) {
    let length = 5  // Simplified length calculation
    let upper = "UPPER"  // Simplified uppercase
    let lower = "lower"  // Simplified lowercase
    return length, upper, lower
}

// Execute tests
let sum, product = calculate_pair(6, 7)
print("Calculations: sum=" + string(sum) + ", product=" + string(product))

let len, up, low = process_string("test")
print("String processing: len=" + string(len) + ", up=" + up + ", low=" + low)
