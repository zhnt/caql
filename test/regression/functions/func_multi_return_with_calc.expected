Calculations: sum=13, product=42
String processing: len=lower, up=UPPER, low=lower
